kernel/Kconfig

```diff
@@ -474,6 +474,56 @@ endmenu
 
 menu "Other Kernel Object Options"
 
+choice SEM_IMPL
+	prompt "Semaphore implementation"
+	default SEM_IMPL_SINGLE_SPINLOCK if !SMP
+	default SEM_IMPL_ATOMIC if SMP
```

Comment:
You want to disable when when CONFIG_ATOMIC_OPERATIONS_C.  Surely @ruuddw  can add detail, but ARC has the disadvantage of being an SMP platform without architectural atomic primitives.  So they emulate atomics using interrupt masking (or, in userspace processes, syscalls!).    Obviously those two techniques will collide badly, and ARC will never want SEM_IMPL_ATOMIC.

Which seems like a good way to get default coverage on SEM_IMPL_SPINLOCK_PER_SEMAPHORE, which seems like the best choice on multiprocessor ARC.

Reply:
Sure, I can do that. I actually dropped a note in the config about that... but didn't know what to put in here for the Kconfig flag for this.

- [ ] default to `SEM_IMPL_SPINLOCK_PER_SEMAPHORE` if SMP and CONFIG_ATOMIC_OPERATIONS_C

Reply:
The SMP atomics issue is temporary (almost resolved). The MetaWare compiler used some older semantics for some compiler generated atomic function, next release fixes that. I wouldn't worry too much about that.
The lack-of-atomics issue is for single core ARC EM (we don't do SMP for this small MCU). Besides for SMP, atomics where used for Userspace (futex).